examples/js/loaders/LDrawLoader.js

```diff
@@ -566,9 +644,66 @@ THREE.LDrawLoader = ( function () {
 
 						if ( parseScope.conditionalSegments.length > 0 ) {
 
-							var lines = createObject( parseScope.conditionalSegments, 2 );
+							var conditionalSegments = parseScope.conditionalSegments;
+							var lines = createObject( conditionalSegments, 2 );
 							lines.isConditionalLine = true;
-							lines.visible = false;
+
+							var controlArray0 = new Float32Array( conditionalSegments.length * 3 * 2 );
+							var controlArray1 = new Float32Array( conditionalSegments.length * 3 * 2 );
+							var directionArray = new Float32Array( conditionalSegments.length * 3 * 2 );
+							for ( var i = 0, l = conditionalSegments.length; i < l; i ++ ) {
+
+								var os = conditionalSegments[ i ];
+								var c0 = os.c0;
+								var c1 = os.c1;
+								var v0 = os.v0;
+								var v1 = os.v1;
+								var index = i * 3 * 2;
+								controlArray0[ index + 0 ] = c0.x;
+								controlArray0[ index + 1 ] = c0.y;
+								controlArray0[ index + 2 ] = c0.z;
+								controlArray0[ index + 3 ] = c0.x;
+								controlArray0[ index + 4 ] = c0.y;
+								controlArray0[ index + 5 ] = c0.z;
+
+								controlArray1[ index + 0 ] = c1.x;
+								controlArray1[ index + 1 ] = c1.y;
+								controlArray1[ index + 2 ] = c1.z;
+								controlArray1[ index + 3 ] = c1.x;
+								controlArray1[ index + 4 ] = c1.y;
+								controlArray1[ index + 5 ] = c1.z;
+
+								directionArray[ index + 0 ] = v1.x - v0.x;
+								directionArray[ index + 1 ] = v1.y - v0.y;
+								directionArray[ index + 2 ] = v1.z - v0.z;
+								directionArray[ index + 3 ] = v1.x - v0.x;
+								directionArray[ index + 4 ] = v1.y - v0.y;
+								directionArray[ index + 5 ] = v1.z - v0.z;
+
+							}
+
+							lines.geometry.addAttribute( 'control0', new THREE.BufferAttribute( controlArray0, 3, false ) );
+							lines.geometry.addAttribute( 'control1', new THREE.BufferAttribute( controlArray1, 3, false ) );
+							lines.geometry.addAttribute( 'direction', new THREE.BufferAttribute( directionArray, 3, false ) );
+
+							lines.material = lines.material.map( mat => {
+
+								return new THREE.ShaderMaterial( {
```

Comment:
@yomboprime Do you have any suggested ways for how to handle this unique type of material? I suppose the right way would be for the material libraries to be refactored to support two different types of line materials.

Reply:
Yes, the library should provide optional lines materials, otherwise you can end up with lots of different custom material shaders with the same colour.

I would replicate `material.userData.edgeMaterial` into another `material.userData.optionalEdgeMaterial` variable. Both are created at the same time with the same colour; one is `LineBasicMaterial` and the other is your custom shader. And wether one or the other is used depends on wether we're currently creating normal lines or optional lines.

Or even better, I would optimize further and create the line materials in a lazy way: Create LineBasic or custom shader line material the first time it is needed if it is null while parsing elements, and set it in the `material.userData`. Probably you will need to store a material's `userdata.edgeColour` when creating it, for the lazy creation.

Reply:
@gkjohnson I can do it myself in another PR if you find it confusing.

Reply:
@yomboprime Thanks! If you already have an idea of how you want to implement the material cache I'd be happy to pass it off. Otherwise I may be able to get to it later in the week / next weekend!

Reply:
All right. This can be merged then.